nck/utils/date_handler.py

```diff
@@ -63,3 +63,24 @@ def get_date_start_and_date_stop_from_date_range(date_range: str) -> Tuple[date,
     """
     current_date = date.today()
     return DEFAULT_DATE_RANGE_FUNCTIONS[date_range](current_date)
+
+
+def build_date_range(start_date, end_date, date_range):
+    """Returns date start and date end adapted if there is a date range.
+
+    Args:
+        start_date (date): start date
+        end_date (date): end date
+        date_range (str): One of the default date ranges that exist
+
+    Returns:
+        Tuple[datetime, datetime]: date start and date stop that match the date range
+    """
+    check_date_range_definition_conformity(start_date, end_date, date_range)
+
+    if date_range is not None:
+        start_date, end_date = get_date_start_and_date_stop_from_date_range(date_range)
+        start_date = datetime.combine(start_date, datetime.min.time())
```

Comment:
Your solution is great, i'm just looking at the details here since it's the only thing new is the convertion from date to datetime. Wouldn't it be more explicit to do something like this: 
```suggestion
        start_date = datetime(year=start_date.year, month=start_date.month, day=start_date.day)
```
Ok it takes a bit less characters to write your option but it is also less explicit to me. I may be wrong though, maybe someone more experienced like @benoitgoujon or @gabrielleberanger have a better vision of both solutions or even come with another one.

Reply:
I agree with you @tom-grivaud. The code works well and uses the properties of the standard library elegantly but your suggestion is indeed a bit easier to understand at first sight. Btw, we don't need to repeat ourselves with args, we can just write:

```suggestion
        start_date = datetime(start_date.year, start_date.month, start_date.day)
```

Reply:
Indeed, it's cleaner that way